plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/project/NewProjectWizard.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.wizard.project;
+
+import java.util.List;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.osgi.util.NLS;
+import org.jboss.tools.common.ui.WizardUtils;
+import org.jboss.tools.openshift.common.ui.wizard.AbstractOpenShiftWizard;
+import org.jboss.tools.openshift.core.connection.Connection;
+import org.jboss.tools.openshift.internal.common.core.job.AbstractDelegatingMonitorJob;
+import org.jboss.tools.openshift.internal.ui.OpenShiftUIActivator;
+
+import com.openshift.restclient.OpenShiftException;
+import com.openshift.restclient.model.IProject;
+
+
+/**
+ * @author jeff.cantrill
+ */
+public class NewProjectWizard extends AbstractOpenShiftWizard<NewProjectWizardModel> {
+
+	public NewProjectWizard(Connection connection, List<IProject> projects) {
+		super("Create OpenShift Project", new NewProjectWizardModel(connection, projects));
+	}
+
+	@Override
+	public boolean performFinish() {
+		AbstractDelegatingMonitorJob job = 
+				new AbstractDelegatingMonitorJob(NLS.bind("Creating project {0}...", getModel().getProjectName())) {
+			@Override
+			protected IStatus doRun(IProgressMonitor monitor) {
+				try {
+					getModel().createProject();
+					return Status.OK_STATUS;
+				} catch (OpenShiftException e) {
+					return new Status(Status.ERROR, 
+							OpenShiftUIActivator.PLUGIN_ID, 
+							NLS.bind("Could not create project \"{0}\": {1}", getModel().getProjectName(), e.getMessage()), e);
+				}
+			}
+		};
+		
+		try {
+			WizardUtils.runInWizard(job, getContainer());
+		} catch (Exception e) {
```

Comment:
I think that catch (InvocationTargetException | InterruptedException e) would be better since we swallow everything here (which then might be keep maintainers wondering where the heck the exception was swallowed)